### PR TITLE
Two more Guppy parameters exposed to user

### LINKED ISSUE #10: Expose guppy parameters to the user
We found out that we had to tweak the `--gpu_runners_per_device` parameter to guppy so it would be good to enable user to do this as this parameter is highly specific to what type of GPU you have. Should probably do this for other guppy parameters as well such as `--cpu_threads_per_caller` and `--num_callers`.

### PR DESCRIPTION
**Pull request by alneberg**
Added the parameters `--guppy_gpu_runners_per_device` and `--guppy_cpu_threads_per_caller`.

This fixes #10 

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/nanoseq branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/nanoseq)
 - [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [ ] Make sure your code lints (`nf-core lint .`).
 - [x] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/nanoseq/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

**Comment by drpatelh** [comment]
Amazing! Thanks @alneberg . Could we shorten the names for the parameters please?

Maybe `--guppy_gpu_runners` and `--guppy_cpu_threads`?

**Comment by alneberg** [comment]
That would indeed be a relief! Thought we wanted to keep the original parameter names, but I guess it's enough to mention the original name in the help text. Will do!

File: main.nf [hunk 1]
```diff
@@ -119,6 +121,8 @@ if (!params.skip_demultiplexing) {
     summary['Barcode Kit ID']     = params.barcode_kit ?: 'Unspecified'
     summary['Guppy Config File']  = params.guppy_config ?: 'Unspecified'
     summary['Guppy GPU Mode']     = params.guppy_gpu ? 'Yes' : 'No'
+    summary['Guppy GPU Runners']  = params.guppy_gpu_runners ?: 'Unspecified'
```
Comment (drpatelh) [inline main.nf]:
```suggestion
    summary['Guppy GPU Runners']  = params.guppy_gpu_runners
```

File: main.nf [hunk 2]
```diff
@@ -119,6 +121,8 @@ if (!params.skip_demultiplexing) {
...
+    summary['Guppy CPU Threads']  = params.guppy_cpu_threads ?: 'Unspecified'
```
Comment (drpatelh) [inline main.nf]:
```suggestion
    summary['Guppy CPU Threads']  = params.guppy_cpu_threads
```

Comment (drpatelh) [reply main.nf]: These parameters have default values set so we can just print those, and if they have been customised then those will be printed instead.

File: main.nf [hunk 3]
```diff
@@ -222,7 +226,9 @@ if (!params.skip_demultiplexing) {
         script:
         barcode_kit = params.barcode_kit ? "--barcode_kits $params.barcode_kit" : ""
         config = params.guppy_config ? "--config $params.guppy_config" : "--flowcell $params.flowcell --kit $params.kit"
-        proc_options = params.guppy_gpu ? "--device $params.gpu_device --num_callers $task.cpus --cpu_threads_per_caller 1 --gpu_runners_per_device 6" : "--num_callers 2 --cpu_threads_per_caller ${task.cpus/2}"
+        gpu_runners = params.guppy_gpu_runners ? "--gpu_runners_per_device $params.guppy_gpu_runners" : "--gpu_runners_per_device 6"
```
Comment (drpatelh) [inline main.nf]:
```suggestion
```

File: main.nf [hunk 4]
```diff
@@ -222,7 +226,9 @@ if (!params.skip_demultiplexing) {
...
+        cpu_threads = params.guppy_cpu_threads ? "--cpu_threads_per_caller $params.guppy_cpu_threads" : "--cpu_threads_per_caller 1"
```
Comment (drpatelh) [inline main.nf]:
```suggestion
```

File: main.nf [hunk 5]
```diff
@@ -222,7 +226,9 @@ if (!params.skip_demultiplexing) {
...
+        proc_options = params.guppy_gpu ? "--device $params.gpu_device --num_callers $task.cpus $cpu_threads $gpu_runners" : "--num_callers 2 --cpu_threads_per_caller ${task.cpus/2}"
```
Comment (drpatelh) [inline main.nf]:
```suggestion
        proc_options = params.guppy_gpu ? "--device $params.gpu_device --num_callers $task.cpus --cpu_threads_per_caller $params.guppy_cpu_threads --gpu_runners_per_device $params.guppy_gpu_runners" : "--num_callers 2 --cpu_threads_per_caller ${task.cpus/2}"
```

**Comment by alneberg** [comment]
Completely agree with your suggestions!

File: docs/usage.md [hunk 6]
```diff
@@ -19,6 +19,8 @@
   * [`--barcode_kit`](#--barcode_kit)
   * [`--guppy_config`](#--guppy_config)
   * [`--guppy_gpu`](#--guppy_gpu)
+  * [`--guppy_gpu_runners_per_device`](#--guppy_gpu_runners_per_device)
```
Comment (drpatelh) [inline docs/usage.md]: This has gone back to the old names?

Comment (alneberg) [reply docs/usage.md]: 🤦‍♂ sorry about that. I think they were never changed from the first suggestion. Should be fixed now.

File: docs/usage.md [hunk 7]
```diff
@@ -164,11 +166,17 @@ Kit used to perform the sequencing e.g. `SQK-LSK109`. Not required if `--guppy_c
...
 
 ### `--guppy_config`
-Guppy config file used for basecalling passed with the `--config` parameter. Cannot be used in conjunction with ``--flowcell` and `--kit`.
+Guppy config file used for basecalling passed with the `--config` parameter. Cannot be used in conjunction with `--flowcell` and `--kit`.
 
 ### `--guppy_gpu`
 Whether to demultiplex with Guppy in GPU mode.
 
+### `--guppy_gpu_runners`
+Number of '--gpu_runners_per_device' used for guppy when using '--guppy_gpu' (default: 6)
```
Comment (drpatelh) [inline docs/usage.md]:
```suggestion
Number of '--gpu_runners_per_device' used for guppy when using `--guppy_gpu` (default: 6)
```

File: docs/usage.md [hunk 8]
```diff
@@ -164,11 +166,17 @@ Kit used to perform the sequencing e.g. `SQK-LSK109`. Not required if `--guppy_c
...
+
+### `--guppy_cpu_threads`
+Number of '--cpu_threads_per_caller' used for guppy when using '--guppy_gpu' (default: 1)
```
Comment (drpatelh) [inline docs/usage.md]:
```suggestion
Number of '--cpu_threads_per_caller' used for guppy when using `--guppy_gpu` (default: 1)
```

Comment (drpatelh) [reply docs/usage.md]: Just these minor changes now @alneberg :+1:

File: docs/usage.md (hunk 7 above)
Comment (alneberg) [reply docs/usage.md]: Of course!

**Pull request merged**